HIP/hip-850.md

```diff
@@ -0,0 +1,70 @@
+---
+hip: 850
+title: Enhancing Supply Key Functionality for NFT Updates in Treasury Account
+author: Patches (@HGraphPunks)
+type: Standards Track
+category: Core
+needs-council-approval: Yes
+status: Last Call
+last-call-date-time: 2024-04-03T07:00:00Z
+created: 2024-01-02
+discussions-to: https://github.com/hashgraph/hedera-improvement-proposal/discussions/660
+requested-by: HGraphPunks, TierBot
+updated: 2024-03-20
+---
+
+## Abstract
+
+This HIP proposes enhancing the Supply Key functionality within the Hedera ecosystem, specifically enabling the Supply Key to update an NFT metadata field while the NFT is held in the treasury account. This ensures secure and controlled adaptability of NFTs, meeting dynamic needs while maintaining their integrity post-distribution.
+
+## Motivation
+
+Currently, there is a limitation in dynamically updating NFTs in a controlled manner. The proposed enhancement to the Supply Key functionality addresses this gap, allowing for secure and selective updates of NFTs within the treasury account, a feature particularly relevant for various evolving use cases in the Hedera ecosystem.
+
+## Rationale
+
+The enhancement focuses on granting the Supply Key the capability to update the data of an NFT serial number exclusively while it is in the treasury account. This approach ensures security and control, preventing unauthorized modifications once the NFT is distributed.
+
+## User Stories
+
+- As a game developer, I want to update game piece NFTs with player stats during gameplay, ensuring the data becomes immutable once the player regains ownership.
+- As an event organizer, I need to update event ticket NFTs with attendee information while they are in the treasury account, guaranteeing data immutability post-transfer.
+
+## Specification
+
+This HIP requires an update to the Supply Key mechanism, enabling it to modify the metadata of NFTs while they are held in the treasury account. The NFT data should remain immutable once distributed, ensuring integrity and trust in the NFT ecosystem.
```

Comment:
Can we expand on this section?
Which existing transaction are we tapping into. Is it just TokenUpdate or some other flow. We just want to be explicit here.

Reply:
@ty-swirldslabs this part could be expanded on